speech_recognition/__init__.py

```diff
@@ -447,10 +447,12 @@ def snowboy_wait_for_hot_word(self, snowboy_location, snowboy_hot_word_files, so
 
         return b"".join(frames), elapsed_time
 
-    def listen(self, source, timeout=None, phrase_time_limit=None, snowboy_configuration=None):
+    def listen(self, source, timeout=None, phrase_time_limit=None, snowboy_configuration=None, stream=False):
```

Comment:
The indirection here is notably strange, so I thought I'd explain here:

In newer versions of python, if a method contains the `yield` keyword, it _always_ returns a generator. As such, the original implementation (now in `_listen`) always returns a generator, and this wrapper method either returns _that_ generator without modification, or iterates on the generator and returns the only result.

There's an alternate path here, where `_listen` always yields buffers, and `listen` is responsible for merging them into a single payload for `stream=False` -- this would also mean moving the logic for truncating the final non-speech frames, which was _slightly_ less preferable to me.

In either case, users of `stream=True` are going to get a few extra frames of audio. Changing that behavior would require the delaying stream emission for `non_speaking_buffer_count`, and it's a bit of a toss-up as to which is the lower-latency end-to-end solution.